site/content/en/docs/Integration Patterns/high-density-gameservers.md

```diff
@@ -0,0 +1,79 @@
+---
+title: "High Density GameServers"
+linkTitle: "High Density GameServers"
+date: 2021-08-31
+weight: 70
+description: >
+  How to run multiple concurrent game sessions in a single GameServer process.
+---
+
+{{< alpha title="Allocation Player Filter and Allocation State Filter" gate="PlayerAllocationFilter,StateAllocationFilter" >}}
+
+Depending on the setup and resource requirements of your game server process, sometimes it can be a more economical 
+use of resources to run multiple concurrent game sessions from within a single `GameServer` instance.
+
+The tradeoff here is that this requires more management on behalf of the integrated game server process and external 
+systems, since it works around the common Kubernetes and/or Agones container lifecycle.
+
+Utilising the new allocation `gameServerState` filter as well as the existing ability to edit the 
+`GameServer` Labels at both [allocation time]({{% ref "/docs/Reference/gameserverallocation.md" %}}), and from 
+within the game server process, [via the SDK]({{% ref "/docs/Guides/Client SDKs/_index.md#setlabelkey-value" %}}), 
+means Agones is able to atomically remove a `GameServer` from the list of potentially allocatable 
+`GameServers` at allocation time, and then return it back into the pool of allocatable `GameServers` if and when the 
+game server process deems that is has room to host another game session. 
+
+<a href="../../../diagrams/high-density.puml.png" target="_blank">
+<img src="../../../diagrams/high-density.puml.png" alt="High Density Allocation Diagram" />
+</a>
+
+{{< alert title="Info" color="info">}}
+To watch for Allocation events, there is the initial `GameServer.status.state` change from `Ready` to `Allocated`,
+but it is also useful to know that the value of `GameServer.metadata.annotations["agones.dev/last-allocated"]` will
```

Comment:
Is last allocated something that gets set automatically? I didn't see a clear explanation for how it's used in the UML diagram. Then I found this note which was helpful, but the example below doesn't set it, which makes me wonder how the game server can use it. Can you add a bit of text to explain where this comes from and what values to expect? Are we looking for true/false, a timestamp, something else?

Reply:
Good point! I've added new sections to the sequence diagram, as well as added to the `alert` box as well. PTAL!